src/include/catalog/catalog_accessor.h

```diff
@@ -0,0 +1,280 @@
+  db_oid_t CreateDatabase(const std::string &name);
+
+  /**
+   * Drop all entries in the catalog that belong to the database, including the database entry
+   * @param db the OID of the database to drop
+   * @return true, unless there was no database entry with the given OID
+   */
+  bool DropDatabase(db_oid_t db);
+
+  /**
+   * Sets the search path of namespaces that should be checked when looking up an
+   * index or table by name.
+   * @param namespaces the namespaces to search given in priority order
+   */
+  void SetSearchPath(std::vector<namespace_oid_t> namespaces);
+
+  /**
+   * @return the current default namespace (first one in search path)
+   */
+  namespace_oid_t GetDefaultNamespace() { return search_path_[0]; }
+
+  /**
+   * Given a namespace name, resolve it to the corresponding OID
+   * @param name of the namespace
+   * @return OID of the namespace, INVALID_NAMESPACE_OID if the namespace was not found
+   */
+  namespace_oid_t GetNamespaceOid(const std::string &name);
+
+  /**
+   * Given a namespace name, resolve it to the corresponding OID
+   * @param name of the namespace
+   * @return OID of the namespace, INVALID_NAMESPACE_OID if the namespace was not found
+   */
+  namespace_oid_t CreateNamespace(const std::string &name);
+
+  /**
+   * Drop all entries in the catalog that belong to the namespace, including the namespace entry
+   * @param ns the OID of the namespace to drop
+   * @return true, unless there was no namespace entry with the given OID
+   */
+  bool DropNamespace(namespace_oid_t ns);
+
+  /**
+   * Given a table name, resolve it to the corresponding OID
+   * @param name of the table
+   * @return OID of the table, INVALID_TABLE_OID if the table was not found
+   */
+  table_oid_t GetTableOid(const std::string &name);
+
+  /**
+   * Given a table name and its owning namespace, resolve it to the corresponding OID
+   * @param ns in which to search for the table
+   * @param name of the table
+   * @return OID of the table, INVALID_TABLE_OID if the table was not found
+   */
+  table_oid_t GetTableOid(namespace_oid_t ns, const std::string &name);
+
+  /**
+   * Given a table name, create a new table entry in the catalog and assign it an OID. This
+   * function does not instantiate the storage object for the table.
+   * @param ns in which the new table will exist
+   * @param name of the new table
+   * @param schema object describing the new table
+   * @return OID for the table, INVALID_TABLE_OID if the table already exists
+   * @warning The catalog accessor assumes it takes ownership of the schema object
+   * that is passed.  As such, there is no guarantee that the pointer is still
+   * valid when this function returns.  If the caller needs to reference the
+   * schema object after this call, they should use the GetSchema function to
+   * obtain the authoritative schema for this table.
+   */
+  table_oid_t CreateTable(namespace_oid_t ns, const std::string &name, const Schema &schema);
```

Comment:
So to create a table, we need to:
* Call CreateTable with a schema that doesn't have the oids yet
* Then use GetSchema to get a schema with the oids.
* Then instanstiate a SqlTable and pass it into SetTablePointer.
Is that all?

Reply:
At face value, yes.

However, if the schema has any constraints (primary key, uniqueness, etc.) then we also have to define, create, and instantiate the necessary indexes as well.  The catalog will not do this.